ext/date/php_date.c

```diff
@@ -1850,6 +1831,31 @@ static HashTable *date_object_get_properties_for(zend_object *object, zend_prop_
 		}
 		zend_hash_str_update(props, "timezone", sizeof("timezone")-1, &zv);
 	}
+}
+
+static HashTable *date_object_get_properties_for(zend_object *object, zend_prop_purpose purpose) /* {{{ */
+{
+	HashTable *props;
+	php_date_obj *dateobj;
+
+	switch (purpose) {
+		case ZEND_PROP_PURPOSE_DEBUG:
+		case ZEND_PROP_PURPOSE_SERIALIZE:
+		case ZEND_PROP_PURPOSE_VAR_EXPORT:
+		case ZEND_PROP_PURPOSE_JSON:
+		case ZEND_PROP_PURPOSE_ARRAY_CAST:
+			break;
+		default:
+			return zend_std_get_properties_for(object, purpose);
+	}
+
```

Comment:
An idea for later, it seems similar code to this is present in other functions, maybe extract this into it's own function?